git-pr-chain.py

```diff
@@ -701,6 +721,62 @@ def cmd_merge(args):
     # PR!
 
 
+def cmd_new_pr(args):
+    """
+    Use this command to mark the top commit as a new PR in the chain. It will
+    generate the PR branch automatically
+    """
+
+    def generate_pr_chain_annotation(commit_title) -> str:
+        downcased = commit_title.lower()
+        # Change any non-words to _
+        only_word_char = re.sub(r"[^\w]", "_", downcased)
+        # Make _ compact: ____ -> _
+        truncate_continuous_underscore = re.sub(r"_+", "_", only_word_char)
+        # Strip _ away from the head and tail: ___xx_xx___ -> xx_xx
+        strip_start_end_underscore = re.sub(
+            r"^_*|_*$", "", truncate_continuous_underscore
+        )
+        # Only keep 40 characters, padding 4 random hex characters at the end
+        random_padding = "".join(random.choice(string.hexdigits) for n in range(4))
+        return f"{strip_start_end_underscore[0:40]}_{random_padding.lower()}"
```

Comment:
It'll now generate a branch name like. The 4 character padding should provide it enough uniqueness, I think.
```
    git-pr-chain: add_hex_padding_to_branch_name_efa2
```

Reply:
I agree.

Just to show my work: 4 hex digits is 2^32 possible nonces.  To get a collision with probability P, you'd need roughly `n = sqrt(2^33 * P)` commits, https://en.wikipedia.org/wiki/Birthday_problem#Square_approximation.  Setting P = 0.1%, that works out to 2,930 PRs, all starting with the same 40 characters.

That's a lot.